### ISSUE TITLE
Remove ExecutionStateTransition-Exceptions for failed tasks

### ISSUE THREAD

**Issue by rmetzger** [stratosphere/stratosphere#484]
The JobManager logfile often contains exceptions like the one attached for failing tasks.

The exception is confusing users since they assume a bug in the system. It is actually not a real bug in the system, its the inability of the ExecutionGraph to properly handle asynchronous tasks.

```
21:59:03,350 INFO  eu.stratosphere.nephele.execution.ExecutionStateTransition    - JM
: ExecutionState set from CANCELING to FAILED for task Map(ItemVectorMapper) (6/9)
21:59:03,350 ERROR eu.stratosphere.nephele.execution.ExecutionStateTransition    - java.lang.IllegalStateException: Unexpected state change: CANCELING -> FAILED
        at eu.stratosphere.nephele.execution.ExecutionStateTransition.checkTransition(ExecutionStateTransition.java:164)
        at eu.stratosphere.nephele.executiongraph.ExecutionVertex.updateExecutionState(ExecutionVertex.java:381)
        at eu.stratosphere.nephele.executiongraph.ExecutionVertex$1.run(ExecutionVertex.java:316)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1170)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:640)
        at java.lang.Thread.run(Thread.java:853)
```

Has anybody a suggestion how to solve it?
Just remove the exception for this particular state transition?

**Comment by StephanEwen** [comment]
I think there is no such thing as an illegal state transition. Certainly
not when the target state is 'failed'. The patch should make sure no
transition to 'failed' is ever treated as unexpected. Let's conservatively
keep the other checks in place for now.
Am 17.02.2014 17:59 schrieb "Robert Metzger" notifications@github.com:

**Comment by StephanEwen** [comment]
Fixed in cd07ed6fd9128655e8ad5b13bc1fe496f538e3ed
